### PR TITLE
Chain the execution node start response promise to the node lifetime

### PR DESCRIPTION
**Pull request by Dakostu**
This change cleans up the response promise lifetime semantics a bit and also fixes a missing retry delay mechanism.

### CODE REVIEW

File: contrib/tenzir-plugins
Comment (dominiklohmann) [inline contrib/tenzir-plugins:1]: This needs a bug fix changelog entry.

File: libtenzir/src/pipeline_executor.cpp [hunk 1]
```diff
@@ -62,10 +62,7 @@ void pipeline_executor_state::start_nodes_if_all_spawned() {
         finish_start();
       },
       [this](const caf::error& err) mutable {
-        if (err == caf::sec::broken_promise) {
-          abort_start(ec::silent);
-          return;
-        }
+        TENZIR_ASSERT(err != caf::sec::broken_promise);
```
Comment (dominiklohmann) [inline libtenzir/src/pipeline_executor.cpp:65]: That's too harsh—what if an operator exits with a broken promise?

File: contrib/tenzir-plugins
Comment (dominiklohmann) [inline contrib/tenzir-plugins:1]: @tobim can you assess this part? I give my approval on the exec node and executor changes, but I'm not deep enough in this right now.

File: libtenzir/src/pipeline_executor.cpp (hunk 1 above)
Comment (tobim) [reply libtenzir/src/pipeline_executor.cpp:65]: Agreed.

FWIW: I still get the `broken promise` diagnostic in the pipeline overview with this branch, the actual error is printed on stderr:
```
*** unhandled exception: [id: 362, name: user.scheduled-actor, exception typeid: boost.wrapexcept]: bind: Address already in use [system:98 at include/boost/asio/detail/reactive_socket_service.hpp:161:33 in function 'boost::system::error_code boost::asio::detail::reactive_socket_service<boost::asio::ip::tcp>::bind(implementation_type &, const endpoint_type &, boost::system::error_code &) [Protocol = boost::asio::ip::tcp]']
```
Pretty ugly, but at least it contains "Address already in use".

Comment (tobim) [reply libtenzir/src/pipeline_executor.cpp:65]: Hm, I think this is a local problem in `tcp-listen`.

Comment (dominiklohmann) [reply libtenzir/src/pipeline_executor.cpp:65]: I'll create a separate PR to generally catch exceptions in operators to turn them into diagnostics.

File: contrib/tenzir-plugins
Comment (tobim) [reply contrib/tenzir-plugins:1]: The retry delay itself is fixed now.

**Review by tobim — APPROVED** [review]

File: contrib/tenzir-plugins
Comment (Dakostu) [reply contrib/tenzir-plugins:1]: I added a changelog entry.

File: libtenzir/src/pipeline_executor.cpp (hunk 1 above)
Comment (Dakostu) [reply libtenzir/src/pipeline_executor.cpp]: I removed the assetion.

Comment (dominiklohmann) [reply libtenzir/src/pipeline_executor.cpp]: Great, think this can be merged then. I'll open a follow-up PR shortly.

File: contrib/tenzir-plugins
Comment (Dakostu) [reply contrib/tenzir-plugins:1]: I added an `early silent abort` integration test and one more edge case check. Should work now.
